opennmt/utils/hooks.py

```diff
@@ -205,7 +205,7 @@ def begin(self):
 
     tf_vars = []
     current_scope = tf.get_variable_scope()
-    with tf.variable_scope(current_scope, reuse=True):
+    with tf.variable_scope(current_scope, reuse=tf.AUTO_REUSE):
```

Comment:
> I would like this code to work with older TensorFlow version, so could you add a fallback for this line:
> 
> Suggested change Beta 
> [Give us feedback](https://www.research.net/r/SuggestedChanges)
>     with tf.variable_scope(current_scope, reuse=tf.AUTO_REUSE):
>     with tf.variable_scope(current_scope, reuse=tf.AUTO_REUSE if hasattr(tf, "AUTO_REUSE") else True):

support TensorFlow < 1.4 which there is no `tf.AUTO_REUSE` now. Maybe we could give up the support of TensorFlow < 1.4 (as said the README.md)  someday.   It will make the code shorter.